algo/src/main/java/com/hartwig/serve/sources/ckb/region/CkbRegion.java

```diff
@@ -0,0 +1,40 @@
+package com.hartwig.serve.sources.ckb.region;
+
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import com.hartwig.serve.ckb.datamodel.clinicaltrial.Location;
+
+import org.immutables.value.Value;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+@Value.Immutable
+@Value.Style(passAnnotations = { NotNull.class, Nullable.class })
+public abstract class CkbRegion {
+
+    @NotNull
+    public abstract String country();
+
+    @NotNull
+    public abstract Set<String> states();
+
+    public boolean includes(@NotNull Location location) {
+        return country().equals(toLowercaseNullable(location.country())) && (states().isEmpty() || states().contains(toLowercaseNullable(
```

Comment:
Minor but I would do this in 2 lines (first line would be `boolean isStateMatch`

Reply:
agrees that is a readability improvement.